Journeys/Email integration/Promise date change communication.md

```diff
@@ -0,0 +1,86 @@
+<service name="bulkSendPromiseDateChangedNotification" engine="java"
+             location="co.hotwax.customerservice.order.CsrOrderServices" invoke="bulkSendPromiseDateChangedNotification" auth="true" transaction-timeout="3600">
+        <description>Bulk service to send promised date changed notification. Default send promised date changed on the last day.</description>
+        <attribute name="fromDate" type="Timestamp" mode="IN" optional="true"/>
+        <attribute name="toDate" type="Timestamp" mode="IN" optional="true"/>
+</service>
+```
+| Parameter | Description | Required |
+|-----------|-------------|----------|
+| `fromDate` | The start time for the service to read the change | No |
+| `toDate` | The end time for the service to read the change| No |
+
+### Step 2: Prepare for Email Notification
+
+The BulkSendPromiseDateChange service picks up all the updated sales order items. This service processes the data and prepares it for the email notification by bringing all the sales order items whose promise dates have changed between `fromDate` and `toDate`.
+
+### Step 3: Call `sendDeliveryDateChangedNotification` service
+After bringing all the updated items in response, the BulkSendPromiseDateChange service calls the `sendDeliveryDateChangedNotification` service for each item.
+
+#### Service Details:
+```
+<service name="sendDeliveryDateChangedNotification" engine="java" require-new-transaction="true" max-retry="3"
+             location="co.hotwax.warehouse.shipping.WarehouseServices" invoke="sendDeliveryDateChangedNotification">
+        <description>Send an order item Delivery Date Changed notification</description>
+        <implements service="orderNotificationInterface"/>
+        <attribute name="orderId" type="String" mode="IN" optional="true"/>
+        <attribute name="orderItemSeqId" type="String" mode="IN" optional="true"/>
+</service>
+```
+| Parameter | Description | Required |
+|-----------|-------------|----------|
+| `orderId` | The ID of the order in Shopify | No |
+| `orderItemSeqId` | The ID of the sequence of an item in the order | No |
+
+### Step 4: Customize Email Notification
+
+The `sendDeliveryDateChangedNotification` service checks the configuration for the email integration platform and picks the appropriate API template. The service then takes input parameters  `orderId` and `orderItemSequenceID` to update all the sales order items in the API template with changed promised dates.
+
+HotWax Commerce has ready integration with Klaviyo, a marketing automation platform. Here's a sample JSON file that is shared with Klaviyo's API for each order:
+```
+{
+  "brandName": "",
+  "DISCOUNT_CODE": "",
+  "DISCOUNT_VALUE": "",
+  "ORDERNO": "",
+  "DISCOUNT_CODE": "",
+  "FIRSTNAME": "",
+  "ITEMLIST_XML": [
+    {
+      "itemId": "",
+      "status": "",
+      "description": "",
+      "quantity": "",
+      "color": "",
+      "price": "",
+      "itemImage": "",
+      "itemUrl": "",
+      "backOrder": "",
```

Comment:
is there no parameter for pre-order?

Reply:
There is a parameter for pre-order as well, this template is configured as per client requirement, if needed it can be changed to backorder or if needed 'pre-order' can also be extended.

Reply:
Discuss this with Deepak, there should be one template that works for everyone because the data is the same. Make a ticket in the OMS git accordingly and update this document with what the fields should be not what they are today.

Reply:
This parameters is not in use, earlier it was used to show some tag at item level on email.
As per current implementation the value of this field is empty.
We can ignore or remove this from documentation

Reply:
> Discuss this with Deepak, there should be one template that works for everyone because the data is the same. Make a ticket in the OMS git accordingly and update this document with what the fields should be not what they are today.

This is the single template used for both preorder/backorder

Reply:
How do we know which one it is? Is this a Y/N flag?

Reply:
Right now we are not sending anything on backorder field

Reply:
> implementation

@dt2patel

Reply:
> This parameters is not in use, earlier it was used to show some tag at item level on email. As per current implementation the value of this field is empty. We can ignore or remove this from documentation

@dt2patel